src/collections/vec.rs

```diff
@@ -1778,12 +1778,43 @@ impl<'bump, T: 'bump + Clone> Vec<'bump, T> {
 }
 
 impl<'bump, T: 'bump + Copy> Vec<'bump, T> {
+    /// Helper method to copy all of the items in `other` and append them to the end of `self`.
+    ///
+    /// SAFETY:
+    ///   * The caller is responsible for:
+    ///       * calling `reserve` beforehand to guarantee that there is enough capacity to
+    ///         store `other.len()` more items.
+    ///       * updating the length of the `Vec` afterward to reflect the number of items added.
+    ///   * The caller must also guarantee that `self` and `other` do not overlap.
+    unsafe fn extend_from_slice_copy_unchecked(&mut self, other: &[T]) {
+        let old_len = self.len();
+        debug_assert!(old_len + other.len() <= self.capacity());
+
+        // SAFETY:
+        // * `src` is valid for reads of `other.len()` values by virtue of being a `&[T]`.
+        // * `dst` is valid for writes of `other.len()` bytes because the caller of this
+        //   method is required to `reserve` capacity to store at least `other.len()` items
+        //   beforehand.
```

Comment:
Can we debug assert this property? Something like

```
debug_assert!(self.capacity() - self.len() >= other.len());
```

Reply:
I added this `debug_assert!` and then later realized that it was equivalent to the one already on line 1790 above the `SAFETY` comment. I removed it again in commit https://github.com/fitzgen/bumpalo/pull/240/commits/60b1407608fdc1c98bf64477ee5bc10facd692fb.